### ISSUE TITLE
Add Slope/Intercept

### ISSUE THREAD

**Issue by rcrum003** [rcrum003/HiveControl#12]
Add the ability for a user to set Slope/Intercept:

Same a for the scale assuming y=mx+b. User enters low engineering unit, prog reads & records raw value, user enters high engineering unit, prog reads & records high raw value.
Slope & offset calculated & displayed. Used may choose not to enter a new low or high value in which case it continues to use last value, this is for single point calibration.

**Comment by NapaOgden** [comment]
Add slope & intercept wizard for humidity.
Sensors have large variation out of the box.

Add slope (gain factor) wizard for Lux sensor, it gets obscured when placed behind a protective cover in the field. Approximately X6 for half ping-pong ball opacity.

weight sensor looses calibration values when disabled, bummer!

**Comment by nanderson77** [comment]
That's what happened!!

I disabled weight a couple days ago to see if it was causing dht22 issues.
Now I'm getting weight out of range errors.

It would be great to retain sensor settings when disabled.  I noticed the
same thing for weather up.  It reset when I switched to dht22 for
troubleshooting.

It would also help just to default to some values that typically read at
least close to a good value on initialization.

**Comment by nanderson77** [comment]
Working on Nowther_01

For bad weights, The message log reads: Raw weight was which exceeded 1500 or was less than 0 setting value to 0.  I reads like it’s supposed to insert the actual weight read, but it doesn’t, it would be helpful to see the current value.

Please make that calibration wizard popup screen bigger.  It’s really easy to miss the save to DB button because you have to scroll down to see it.

Even after hitting save to DB and getting the successful message… these calibration values are not being updated reliably.  I saved to DB three times before it actually did save the values. 

Lastly, even after the calibration wizard says the current calibration values will give a value of 124.4 I’m still getting the out of range error and of course constant 0 for weight

I didn’t do a screenshot of the cal values because I couldn’t fit everyting in the window – but copy/paste of text is just as good.  My current zero and calibration slope match the values shown.

Zero Value-4972

Known Weight124.4

Value for Known Weight (we removed zero already):1196343

Solving for calibration valueCalibration Should be:9616.9051446945

Reading the scale with the recommended zero and calibration would give us a weight of:124.4

NOTE: scale values will vary up to .5 lb per the specifications of the load cells (more or less depending on your equipment.

The top of my list of requests is a better present value screen– something with all of the raw and scaled values that I can manually refresh, or the screen can refresh based on some refresh rate we enter.  

Nate Anderson

Home 763-537-4193

Cell 763-688-0432

Skype 763-634-0115 (nanderson777)

**Comment by rcrum003** [comment]
Are you trying to calibrate with the scale under the hive?

If so, this won't work based on the calc we do.

We need to see what zero would look like.

**Comment by nanderson77** [comment]
I completely unload the load cells by putting shims between the scale and the base for collecting the zero value.  I then take out the shims and capture the loaded weight.

Nate Anderson

Home 763-537-4193

Cell 763-688-0432

Skype 763-634-0115 (nanderson777)

**Comment by rcrum003** [comment]
Ok, i figured out your weight problem.

In /etc/hosts, you had your hostname misspelled (was missing the _)
So when sudo fired off (need sudo to talk with the GPIOs) - it was giving a sudo error when pulling the raw weight, thus, corrupting the initial RAW value we used to calculate everything else….

I’ve fixed that for you.

Ryan

**Comment by nanderson77** [comment]
Isn’t that strange I wonder why it picked last week to pop up?  I’ve never opened that file before.  I changed the hostname several weeks ago – but had no idea I needed to change the value in the hosts file.  

Does evidence of that misconfiguration show up in any of the log files I should be looking at, or is it a real oddity?  

Nate Anderson

Home 763-537-4193

Cell 763-688-0432

Skype 763-634-0115 (nanderson777)
